README.md

```diff
@@ -1,61 +1,107 @@
+You can use this in combination with a script that periodically refreshes your token.
+
+### Usage in FreeStyle Jobs
+If you still use free style jobs (hint: you should consider migrating to [Jenkinsfile](https://jenkins.io/doc/book/pipeline/)), you can configure both configuration and the secrets you need on the job level.
+
+![Job Configuration](docs/images/job_screen.png)
+
+The secrets are available as environment variables then. 
+
+### Usage via Jenkinsfile
+Let the code speak for itself:
 ```groovy
 node {
+  // define the secrets and the env variables
   def secrets = [
       [$class: 'VaultSecret', path: 'secret/testing', secretValues: [
           [$class: 'VaultSecretValue', envVar: 'testing', vaultKey: 'value_one'],
           [$class: 'VaultSecretValue', envVar: 'testing_again', vaultKey: 'value_two']]],
       [$class: 'VaultSecret', path: 'secret/another_test', secretValues: [
           [$class: 'VaultSecretValue', envVar: 'another_test', vaultKey: 'value']]]
   ]
-
-  wrap([$class: 'VaultBuildWrapper', vaultSecrets: secrets]) {
+  
+  // optional configuration, if you do not provide this the next higher configuration
+  // (e.g. folder or global) will be used
+  def configuration = [$class: 'VaultConfiguration', 
+                       vaultUrl: 'http://my-very-other-vault-url.com', 
+                       vaultCredentialId: 'my-vault-cred-id']
+  
+  // inside this block your credentials will be available as env variables
+  wrap([$class: 'VaultBuildWrapper', configuration: configuration, vaultSecrets: secrets]) {
       sh 'echo $testing'
       sh 'echo $testing_again'
       sh 'echo $another_test'
   }
 }
 ```
+In the future we might migrate to a [BuildStep](http://javadoc.jenkins-ci.org/hudson/tasks/BuildStep.html) instead of a BuildWrapper.
 
-## TODO
-
-* [JENKINS-37203 - Override Authentication Token?" do not work](https://issues.jenkins-ci.org/browse/JENKINS-37203)
-* Add a [BuildStep](http://javadoc.jenkins-ci.org/hudson/tasks/BuildStep.html) for use as an alternative to the Build Wrapper.
-
-## CHANGELOG
+# Migration Guide
+### Upgrade from 1.x to 2.0
+The `BuildWrapper` did not change, so no changes to your Jenkinsfile should be necessary. However, you need to reconfigure Vault in your Jenkins instance based on the instructions above. There is no way to smoothly upgrade this, because this is a major rewrite and handling of configuration completly changed.
+# CHANGELOG
 
-* **2017/04/10** - Feature Release - 1.4
-  * Support reading Vault Token from file on disk [JENKINS-37713](issues.jenkins-ci.org/browse/JENKINS-37713)
-  * Using credentials plugin for authentication token [JENKINS-38646](issues.jenkins-ci.org/browse/JENKINS-38646)
-* **2017/03/03** - Feature Release - 1.3
-  * Vault Plugin should mask credentials in build log [JENKINS-39383](issues.jenkins-ci.org/browse/JENKINS-39383)
+* **2017/05/04** - Breaking change release (AppRole auth backend, Folder ability, improved configuration, ...)
```

Comment:
If you can align this with the current master branches changelog that would be great